### ISSUE TITLE
Stalled Check keeps running for obliterated queues

### ISSUE THREAD

**Issue by Mahmo0odsalah** [OptimalBits/bull#2433]
## Description
When obliterating a queue, the Stalled Check keeps running which causes the stalled-check key to be created in Redis. This causes a lot of junk keys when creating a large number of queues. Our use case dictates we create and delete queues on the fly so this is causing our Redis cluster memory to implode.

## Minimal, Working Test code to reproduce the issue.
```
const Queue = require("bull");
const Redis = require("ioredis");
const redis = new Redis();

const queue = new Queue("stalled", {
    prefix: `{${"test-stalled-checker"}}`,
    defaultJobOptions: {
      removeOnComplete: true,
    },
    settings: {
      stalledInterval: 5000,
    },
  });
  queue.process(async (job) => {
    job.progress(50);
  });
  // queue.add({ test: "test" });

  await queue.obliterate();
  console.log(await redis.keys("*test-stalled-checker*"));
  // sleep for 7 seconds
  await new Promise((resolve) => setTimeout(resolve, 7000));
  console.log(await redis.keys("*test-stalled-checker*"));
```

## Bull version
^4.9.0

## Additional information
Our setup uses Redis clusters not a single instance in case this might have anything to do with this issue.

**Comment by manast** [comment]
Are you aware that the stalled keys have an expiration time (default 30s), so they will disappear automatically?

**Comment by manast** [comment]
In other words, if you close your queue after calling obliterate, worst case you will have 1 key left for 30 seconds.

**Comment by Mahmo0odsalah** [comment]
@manast This is not the behavior I'm encountering. Once the key expires another key is immediately created, so the workers keep checking for stalled jobs forever, which is suboptimal because the app could possibly create and obliterate ~100k queues per day.

**Comment by manast** [comment]
They will create the keys as long as you have a Queue instance running. You need to close the instances that are not used anymore and then the keys will disappear. I do not have any other solution, since it is impossible to know if you plan to continue using a queue or not while you still have an instance running.

**Comment by Mahmo0odsalah** [comment]
@manast Do you mean calling `queue.close()` before `queue.obliterate()` ?

**Comment by manast** [comment]
@Mahmo0odsalah You need to call it after, as a closed queue instance will not be usable anymore. Once the queue is closed then the stalled-check will be removed automatically by Redis. If it does not work it means that you still have an instance running somewhere.

**Comment by Mahmo0odsalah** [comment]
@manast I'm afraid this is not working either. 

> If it does not work it means that you still have an instance running somewhere.

How can I know if this is true?

**Comment by Mahmo0odsalah** [comment]
@manast I don't think it's something from my code. Check the code below, I tried clearing the variable so that it gets garbage collected but the issue still persists.

```

  const Queue = require("bull");
  const Redis = require("ioredis");

  let queue = new Queue("stalled", {
    createClient: (type, opts) => {
      return new Redis.Cluster(
        [
          {
            port: process.env.REDIS_PORT || 6379,
            host: process.env.REDIS_URL,
          },
        ],
        {
          retryDelayOnClusterDown: 1000,
          redisOptions: { db: process.env.REDIS_DB || 0 },
        }
      );
    },
    prefix: `{${"stalled-thing"}}`,
    defaultJobOptions: {
      removeOnComplete: true,
    },
    settings: {
      stalledInterval: 2000,
    },
  });

  await queue.obliterate();
  await queue.close(true);
  queue = null;
```

**Comment by manast** [comment]
I do not have a cluster right now to test, but this code is not leaving anything in redis:
```ts
const Queue = require("bull");

async function test_2433() {
  let queue = new Queue("stalled", {
    prefix: `{${"stalled-thing"}}`,
    defaultJobOptions: {
      removeOnComplete: true,
    },
    settings: {
      stalledInterval: 2000,
    },
  });

  await queue.obliterate();
  await queue.close(true);
  queue = null;
}

test_2433();
```

![image](https://user-images.githubusercontent.com/95200/192776001-57f6e4e6-2a21-4151-9d89-c74185eab86a.png)

**Comment by Mahmo0odsalah** [comment]
I found the culprit in my code, the issue was that I was calling `Queue#create` everytime I wanted to access the queue, causing loose references that don't allow the queue to be collected by the garbage collector. Managed to solve it by storing the queues in memory and deleting them on obliteration. I think I wouldn't have made this mistake if there was a `Queue#get` function. 

Feel free to close the issue if there's no plan to implement such a feature. Thanks a lot for your cooperation though :+1:
